generator/internal/dart/darttemplate.go

```diff
@@ -61,13 +61,15 @@ type serviceAnnotations struct {
 }
 
 type messageAnnotation struct {
-	Name             string
-	DocLines         []string
-	ConstructorBody  string // A custom body for the message's constructor.
-	BasicFields      []*api.Field
-	HasFields        bool
-	HasToStringLines bool
-	ToStringLines    []string
+	Name              string
+	DocLines          []string
+	ConstructorBody   string // A custom body for the message's constructor.
+	HasFields         bool
+	HasCustomEncoding bool
+	FromJsonLines     []string
```

Comment:
The FromJson / ToJson info could also be annotated off of the fields of the message.

Reply:
That would make the mustache templates less opaque.  I would like to give the partition idea a try though.